src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1162,21 +1162,55 @@ private static ItemStack handleKeepX(
      */
     public boolean checkOrRequestItems(@Nullable final ItemStack... items)
     {
-        return checkOrRequestItems(true, true, items);
+        return checkOrRequestItems(true, items);
     }
 
     /**
      * Require that items are in the workers inventory.
      * This safeguard ensures you have said items before you execute a task.
      * Please stop execution on false returned.
      *
-     * @param useItemDamage use item damage?
-     * @param items the items needed
+     * @param useItemDamage compare the itemDamage of the values.
+     * @param items         the items needed
      * @return false if they are in inventory
      */
     public boolean checkOrRequestItems(final boolean useItemDamage, @Nullable final ItemStack... items)
     {
-        return checkOrRequestItems(useItemDamage, true, items);
+        if (items == null)
+        {
+            return false;
+        }
+        boolean allClear = true;
+        for (final @Nullable ItemStack tempStack : items)
+        {
+            final ItemStack stack = tempStack.copy();
+            if (stack == null || stack.getItem() == null)
+            {
+                continue;
+            }
+            final int itemDamage = useItemDamage ? stack.getItemDamage() : -1;
+            final int countOfItem = worker.getItemCountInInventory(stack.getItem(), itemDamage);
+
+            if (countOfItem < 1)
+            {
+                final int itemsLeft = stack.stackSize - countOfItem;
+                @NotNull final ItemStack requiredStack = new ItemStack(stack.getItem(), itemsLeft, itemDamage);
+                getOwnBuilding().addNeededItems(requiredStack);
+                allClear = false;
+            }
+            else
+            {
+                getOwnBuilding().clearNeededItems();
```

Comment:
Also do we want to clear all the needed items, or if only remove the current item if we have it?